aries-site/src/layouts/content/SubsectionHeader.js

```diff
@@ -1,16 +1,46 @@
-import React, { useState } from 'react';
+import React, { useState, useContext, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { Box, Button, Heading } from 'grommet';
 import { Link as LinkIcon } from 'grommet-icons';
 
+import { ThemeContext } from 'styled-components';
+import { useIntersection } from '@hpe/react-hooks';
 import { HighlightPhrase } from '../../components';
 import { nameToSlug } from '../../utils';
 
+import { ActiveHeadingContext } from './ActiveHeadingContext';
+
 export const SubsectionHeader = ({ children, level }) => {
   const [over, setOver] = useState(false);
   const id = nameToSlug(children);
+  const { setActiveHeading } = useContext(ActiveHeadingContext);
+
+  const theme = useContext(ThemeContext);
+  let { large } = theme.global.edgeSize;
+  large = parseInt(large.replace('px', ''), 10); // 48
+
+  // top and bottom margin values to calculate intersection window
+  const topMargin = large;
+  const bottomMargin = `-${2 * large}`;
+
+  const options = {
+    rootMargin: `${topMargin}% 0% ${bottomMargin}% 0%`,
+    // threshold: 1.0,
+  };
+
+  const [headingToWatch, entry] = useIntersection({ options });
```

Comment:
This isn't quite making sense to me. Isn't the first thing returned the `setNode` function so you can set the node state in the hook? I would think you'd need to call this function with the ref.current in a useEffect rather than pass it like a reference on line 43. Or am I missing something?

Reply:
I went off of the example [here](https://github.com/HewlettPackard/hpe-js-project-scripts/tree/master/packages/react-hooks#useintersection) which is HPE's useIntersection documentation, but I'm not sure if it is outdated/not best practices. In the actual code, node and entry are returned, but does the current implementation accept the ref correctly?
![image](https://user-images.githubusercontent.com/23144209/184954380-41b3dce4-22b8-4ef7-a508-68d80a4fe734.png)

Reply:
Very true. It's a little subtle but it seems like rather than actually using a ref it's relying on React to call the thing passed in as the `ref` like a setter function. e.g. https://flow.org/en/docs/react/refs/

Reply:
Sorry I wasn't able to follow-up yesterday, but is it ok to keep it as is or should I change it to use ref.current in a useEffect?

Reply:
No problem if it's working fine. Feel free to leave it as is.